bulla-contracts/tests/BullaFactoring.test.ts

```diff
@@ -196,16 +196,16 @@ test("it handles BullaFactoring events", () => {
   handleDepositMade(depositMadeEvent);
 
   const depositMadeEventId = getDepositMadeEventId(depositMadeEvent);
-  assert.fieldEquals("DepositMadeEvent", depositMadeEventId, "depositor", depositMadeEvent.params.depositor.toHexString());
+  assert.fieldEquals("DepositMadeEvent", depositMadeEventId, "depositor", depositMadeEvent.params.sender.toHexString());
   assert.fieldEquals("DepositMadeEvent", depositMadeEventId, "assets", depositMadeEvent.params.assets.toString());
-  assert.fieldEquals("DepositMadeEvent", depositMadeEventId, "sharesIssued", depositMadeEvent.params.sharesIssued.toString());
+  assert.fieldEquals("DepositMadeEvent", depositMadeEventId, "sharesIssued", depositMadeEvent.params.shares.toString());
   assert.fieldEquals("DepositMadeEvent", depositMadeEventId, "poolAddress", MOCK_BULLA_FACTORING_ADDRESS.toHexString());
 
   log.info("✅ should create a DepositMade event", []);
 
   const depositMadeWithAttachmentEvent = newDepositMadeWithAttachmentEvent(depositor, assets, shares);
```

Comment:
in theory, the subgraph would not need to know about DepositMadeWithAttachment, since it was the underlying event emitted as well

Reply:
then we should remove those events from the contracts and not handle them separately? On the frontend we differentiate these

Reply:
Actually nvm you are right, in the subgraph we need to keep track of the attachment in the deposit.
But I would only treat the attachment part of the event. Since we will have the Deposit event emitted with it.

So I would only track the Deposit/Withdraw events for the funds, but we look at the attachment events to assigned the attachment to the deposit/withdraw.

Reply:
that's a frontend fix, right?

Reply:
hmm no everything should be in the subgraph I believe.

Reply:
the deposit object in the graph should have a nullable attachment property. Then when the attachment event comes in, we find the deposit/withdrawal associated and set the attachment.

maybe I'm misunderstanding your question.